src/garage/torch/policies/tanh_gaussian_mlp_policy.py

```diff
@@ -0,0 +1,184 @@
+            to avoid numerical issues (plain value - not log or exponentiated).
+        max_std (float): If not None, the std is at most the value of max_std,
+            to avoid numerical issues (plain value - not log or exponentiated).
+        std_parameterization (str): How the std should be parametrized. There
+            are two options:
+            - exp: the logarithm of the std will be stored, and applied a
+               exponential transformation
+            - softplus: the std will be computed as log(1+exp(x))
+        layer_normalization (bool): Bool for using layer normalization or not.
+
+    """
+
+    def __init__(self,
+                 env_spec,
+                 hidden_sizes=(32, 32),
+                 hidden_nonlinearity=nn.ReLU,
+                 hidden_w_init=nn.init.xavier_uniform_,
+                 hidden_b_init=nn.init.zeros_,
+                 output_nonlinearity=None,
+                 output_w_init=nn.init.xavier_uniform_,
+                 output_b_init=nn.init.zeros_,
+                 init_std=1.0,
+                 min_std=np.exp(-20.),
+                 max_std=np.exp(2.),
+                 std_parameterization='exp',
+                 layer_normalization=False):
+
+        self._obs_dim = env_spec.observation_space.flat_dim
+        self._action_dim = env_spec.action_space.flat_dim
+
+        Policy.__init__(self, env_spec, name='TanhGaussianPolicy')
+        TanhGaussianMLPTwoHeadedModule.__init__(
+            self,
+            input_dim=self._obs_dim,
+            output_dim=self._action_dim,
+            hidden_sizes=hidden_sizes,
+            hidden_nonlinearity=hidden_nonlinearity,
+            hidden_w_init=hidden_w_init,
+            hidden_b_init=hidden_b_init,
+            output_nonlinearity=output_nonlinearity,
+            output_w_init=output_w_init,
+            output_b_init=output_b_init,
+            init_std=init_std,
+            min_std=min_std,
+            max_std=max_std,
+            std_parameterization=std_parameterization,
+            layer_normalization=layer_normalization)
+
+    def get_action(self, observation):
+        """Get a single action given an observation.
+
+        Args:
+            observation (np.ndarray): Observation from the environment.
+
+        Returns:
+            tuple:
+                * np.ndarray: Predicted action.
+                * dict:
+                    * list[float]: Mean of the distribution
+                    * list[float]: Standard deviation of logarithmic values of
+                        the distribution
+
+        """
+        with torch.no_grad():
+            if not isinstance(observation, torch.Tensor):
+                observation = tu.from_numpy(observation)
+            observation = torch.Tensor(observation).unsqueeze(0)
+            dist = self.forward(observation)
+            ret_mean = dist.mean.squeeze(0).cpu().numpy()
+            ret_log_std = (dist.variance**.5).log().squeeze(0).cpu().numpy()
```

Comment:
Generally, if you want a square root, you should just call `.sqrt()`.

I thought it would be less efficient to use `**0.5`, but I was surprised to find that pytorch actually will [turn `pow(n, 0.5)` back into sqrt on its own](https://github.com/pytorch/pytorch/blob/7a3ed36309f48cb833f1690991c7b0f59da6ce11/aten/src/ATen/native/cuda/PowKernel.cu#L122). It's still more readable to use `.sqrt()` though.